moe/util/core/match.py

```diff
@@ -34,10 +34,51 @@
 MATCH_TRACK_FIELD_WEIGHTS = {
     "composer": 0.8,
     "disc": 0.3,
+    "duration": 0.8,
     "title": 0.7,
     "track_num": 0.9,
 }  # how much to weigh matches of various fields
 
+# match penalty when both tracks are missing data for a field
+BOTH_MISSING_DATA_PENALTY = 0.1
+
+# match penalty when only one track is missing data for a field
+ONE_MISSING_DATA_PENALTY = 0.2
+
+
+def _duration_penalty(duration_a: float, duration_b: float) -> float:
+    """Returns a penalty value for duration matching.
+
+    Uses a tolerance-based penalty system where a duration mismatch of 2.5% or less
+    is not penalized, and the penalty increases linearly from 0 at 2.5% to 1 at 10%
+    mismatch.
+
+    Args:
+        duration_a: First duration value in seconds.
+        duration_b: Second duration value in seconds.
+
+    Returns:
+        Penalty value between 0.0 and 1.0.
+    """
+    min_penalty_threshold = 0.025
+    max_penalty_threshold = 0.10
+
+    if duration_a <= 0.0 and duration_b <= 0.0:
```

Comment:
It doesn't seem like this check or the ONE_MISSING_DATA_PENALTY code is actually reachable because 0.0 is a False value so it's covered in `get_match_value`. You should be able to remove both conditions and then make the missing_data_penalty variables local to `get_match_value`.